Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,708 @@
+	if (Z == 0) goto branchdestf16;
+}
+
+:BNEQ branchdestb16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=12 & branch_sign=1 ; branchdestb16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (Z == 0) goto branchdestb16;
+}
+
+:BNOV branchdestf16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=10 & branch_sign=0 ; branchdestf16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (O == 0) goto branchdestf16;
+}
+
+:BNOV branchdestb16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=10 & branch_sign=1 ; branchdestb16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (O == 0) goto branchdestb16;
+}
+
+:BOV branchdestf16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=2 & branch_sign=0 ; branchdestf16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (O == 1) goto branchdestf16;
+}
+
+:BOV branchdestb16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=2 & branch_sign=1 ; branchdestb16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (O == 1) goto branchdestb16;
+}
+
+:BPL branchdestf16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=3 & branch_sign=0 ; branchdestf16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (S == 0) goto branchdestf16;
+}
+
+:BPL branchdestb16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=3 & branch_sign=1 ; branchdestb16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (S == 0) goto branchdestb16;
+}
+
+:BUSC branchdestf16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=7 & branch_sign=0 ; branchdestf16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (S != C) goto branchdestf16;
+}
+
+:BUSC branchdestb16 is opcode6_9=0x0008 & branch_external=0 & branch_condition=7 & branch_sign=1 ; branchdestb16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	if (S != C) goto branchdestb16;
+}
+
+:CLRC is opcode0_9=0x0006 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	C = 0;
+}
+
+:CMP addr16, reg0_2 is opcode3_9=0x0068 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	comparison(reg0_2, *:2 addr16);
+}
+
+:CMPR reg3_5, reg0_2 is opcode6_9=0x0005 & reg3_5 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	comparison(reg0_2, reg3_5);
+}
+
+:COMR reg0_2 is opcode3_9=0x0003 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = ~reg0_2;
+	resultFlags(reg0_2);
+}
+
+:DECR reg0_2 is opcode3_9=0x0002 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg0_2 - 1;
+	resultFlags(reg0_2);
+}
+
+:DIS is opcode0_9=0x0003 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 0;
+}
+
+:EIS is opcode0_9=0x0002 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 1;
+}
+
+:GSWD reg0_1 is opcode2_9=0x000C & reg0_1 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local mask:2 = (zext(S) << 7) + (zext(Z) << 6) + (zext(O) << 5) + (zext(C) << 4);
+	reg0_1 = (mask << 16) + mask; 
```

Comment:
This looks odd, isn't it only 16 bits wide making the shift a 0+mask

Reply:
Ooops, this should actually be `reg0_1 = (mask << 8) + mask;` indeed, not 16.